--- a/include/fmt/format-inl.h
+++ b/include/fmt/format-inl.h
@@ -1392,13 +1392,13 @@
     for (auto i = n.bigits_.size(); i > 0; --i) {
       auto value = n.bigits_[i - 1u];
       if (first) {
-        out = format_to(out, FMT_STRING("{:x}"), value);
+        out = format_to(out, runtime("{:x}"), value);
         first = false;
         continue;
       }
-      out = format_to(out, FMT_STRING("{:08x}"), value);
+      out = format_to(out, runtime("{:08x}"), value);
     }
     if (n.exp_ > 0)
-      out = format_to(out, FMT_STRING("p{}"),
+      out = format_to(out, runtime("p{}"),
                       n.exp_ * detail::bigint::bigit_bits);
     return out;
   }

