--- a/pngpriv.h
+++ b/pngpriv.h
@@ -515,7 +515,8 @@
 
 #  if (defined(__MWERKS__) && defined(macintosh)) || defined(applec) || \
-    defined(THINK_C) || defined(__SC__) || defined(TARGET_OS_MAC)
+    defined(THINK_C) || defined(__SC__) || \
+    (defined(TARGET_OS_MAC) && !defined(TARGET_OS_IPHONE))
    /* We need to check that <math.h> hasn't already been included earlier
     * as it seems it doesn't agree with <fp.h>, yet we should really use
     * <fp.h> if possible.

